tests/test_dia_object.py

```diff
@@ -144,6 +147,40 @@ def test_dia_source_append_and_update(self):
         dia_obj.update()
         self._compare_dia_object_values(dia_obj, 0, compare_values)
 
+    def test_dia_source_append_different_schema(self):
+        """ Test the appending of a DIASource to a DIAObject where the DIASource
+        has a more complex schema.
+
+        We also test that the update function works properly and that the
+        summary statistics are computed as expected.
+        """
+        single_source = create_test_dia_sources(1)
+        dia_obj = DIAObject(single_source, None)
+        self.assertEqual(dia_obj.id, single_source[0].getId())
+        self.assertTrue(dia_obj.is_updated)
+
+        extended_schema = make_minimal_dia_source_schema()
+        extended_schema.addField('test', type=float)
+
+        sources = create_test_dia_sources(2, schema=extended_schema)
+        dia_obj.append_dia_source(sources[1])
+        self.assertFalse(dia_obj.is_updated)
+
+        associated_sources = dia_obj.dia_source_catalog
+        self.assertEqual(len(associated_sources), 2)
+        self.assertEqual(associated_sources[-1].getId(),
+                         sources[-1].getId())
+        self.assertEqual(associated_sources[-1].getCoord(),
+                         sources[-1].getCoord())
+
+        compare_values = {
+            "coord_ra": 0.4999619199226218,
+            "coord_dec": 0.5000190382261059
```

Comment:
Where do these numbers come from?

Reply:
I've added a comment. Basically it's just average position of the incremented DIASources that are put into this object.

Reply:
It would be better if you actually computed that average: these are too much "magic value" for my tastes, given that they depend on code that lives elsewhere in this file.

Reply:
Okay, I'll use the original source catalog arrays to compute the values to get rid the magic numbers.